internal/packages/packages.go

```diff
@@ -5,94 +5,63 @@
 package packages
 
 import (
-	"encoding/json"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 
+	"github.com/elastic/go-ucfg"
+	"github.com/elastic/go-ucfg/yaml"
 	"github.com/pkg/errors"
-	"gopkg.in/yaml.v3"
 )
 
 const (
 	// PackageManifestFile is the name of the package's main manifest file.
 	PackageManifestFile = "manifest.yml"
 
-	// DataStreamManifestFile is the name of the dataStream's manifest file.
+	// DataStreamManifestFile is the name of the data stream's manifest file.
 	DataStreamManifestFile = "manifest.yml"
 )
 
-// VarValue represents a variable value as defined in a package or dataStream
-// manifest file.
-type VarValue struct {
-	scalar string
-	list   []string
```

Comment:
I'm curious why you removed `VarValue` and replaced it with the broader `interface{}` type. AFAICT the only types of values allowed for input configuration variables are scalars (int, string, boolean, etc.) or lists of scalars. We don't support maps/dicts. That was the intent behind creating a stricter `VarValue` type that encapsulated these rules.

Reply:
It's because of ucfg which (I suppose) doesn't support UnmarshalYAML and MarshalJSON methods. It seemed to me that the codebase would be smaller, although I'm aware of the risk you brought here.

Reply:
Could you keep `VarValue` and make it implement https://godoc.org/github.com/elastic/go-ucfg#ConfigUnpacker somehow?

Reply:
I can try to do this, although I'm not convinced it will work (didn't see this earlier).

I see the benefits of having stricter `VarValue` type, so if you prefer I can switch back from `ucfg` to our implementation and extend it with same trick for expanding config map keys as in package-spec.

Reply:
Thanks, I appreciate you giving it a try. There are a few examples of implementing `ConfigUnpacker` in the Beats repo. You can find them in your IDE with this regex: `func \(.* Unpack\(.*Config`.

If you succeed, it would allow us to preserve the strictness of what a variable value can be, which is very nice.

If it doesn't work, let's stick with `ucfg` and reduce the strictness of what a variable value can be. I think that's the better tradeoff than implementing some trick to parse flat values with the `yaml` package.

Reply:
I think I got it fixed. Let's see what CI will say.